### DISCUSSION TITLE
*** Could not read served logs: [Errno -3] Temporary failure in name resolution

### DISCUSSION THREAD

**Discussion by manel00** [apache/airflow#40291] [General] [answered]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.7.3

### What happened?

*** Could not read served logs: [Errno -3] Temporary failure in name resolution

In logs in the task in Airflow

![image](https://github.com/apache/airflow/assets/8410687/852b3fe3-92c5-458d-9f46-d4cd1f4f1a05)

### What you think should happen instead?

It is not well connected the worker with the host, because i am trying to retrieve the logs from worker on host

### How to reproduce

having a worker in one server and an airflow host in another

### Operating System

Ubuntu 22

### Deployment

Docker-Compose

### Deployment details

  airflow-worker:
    <<: *airflow-common
    command: >
      bash -c "airflow celery worker -q myqueue --pid=/opt/airflow/airflow-worker.pid"
    healthcheck:
      test:
        - "CMD-SHELL"
        - 'celery --app airflow.executors.celery_executor.app inspect ping -d "celery@$${HOSTNAME}"'
      interval: 10s
      timeout: 10s
      retries: 5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
This is issue with your DNS that you have cofigured for your webserver deployment - it might have not enough resources, have flaky networking or similar. It has nothing to do with Airflow. 

And obligatory haiku:

![image](https://github.com/apache/airflow/assets/595491/00e0feb2-caba-4d6f-9f1c-ffa47b302a90)

**Comment by potiuk** [comment]
(Btw. If you need help with deployment problems, open a discussion not issue).

**Comment by manel00** [comment]
![image](https://github.com/apache/airflow/assets/8410687/cf9ea591-df58-42e2-bfc5-c69b449c9a27)

My service it is this in airflow-host, what am I doing wrong? Thank you

**Reply by potiuk** [reply]
No idea. Make sure that your DNS is stable and has enough resources. It's (again) a deployment issue. Look for DNS problems in the particular deployment type you have (docker-compose I see)

**Comment by manel00** [comment]
That is my traceback from my airflow-host docker-compose:

```
Traceback (most recent call last):
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_exceptions.py", line 10, in map_exceptions
airflow-webserver_1       |     yield
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/backends/sync.py", line 94, in connect_tcp
airflow-webserver_1       |     sock = socket.create_connection(
airflow-webserver_1       |            ^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/usr/local/lib/python3.11/socket.py", line 827, in create_connection
airflow-webserver_1       |     for res in getaddrinfo(host, port, 0, SOCK_STREAM):
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/usr/local/lib/python3.11/socket.py", line 962, in getaddrinfo
airflow-webserver_1       |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       | socket.gaierror: [Errno -3] Temporary failure in name resolution
airflow-webserver_1       | 
airflow-webserver_1       | During handling of the above exception, another exception occurred:
airflow-webserver_1       | 
airflow-webserver_1       | Traceback (most recent call last):
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
airflow-webserver_1       |     yield
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_transports/default.py", line 218, in handle_request
airflow-webserver_1       |     resp = self._pool.handle_request(req)
airflow-webserver_1       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py", line 253, in handle_request
airflow-webserver_1       |     raise exc
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_sync/connection_pool.py", line 237, in handle_request
airflow-webserver_1       |     response = connection.handle_request(request)
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_sync/connection.py", line 86, in handle_request
airflow-webserver_1       |     raise exc
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_sync/connection.py", line 63, in handle_request
airflow-webserver_1       |     stream = self._connect(request)
airflow-webserver_1       |              ^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_sync/connection.py", line 111, in _connect
airflow-webserver_1       |     stream = self._network_backend.connect_tcp(**kwargs)
airflow-webserver_1       |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/backends/sync.py", line 93, in connect_tcp
airflow-webserver_1       |     with map_exceptions(exc_map):
airflow-webserver_1       |   File "/usr/local/lib/python3.11/contextlib.py", line 155, in __exit__
airflow-webserver_1       |     self.gen.throw(typ, value, traceback)
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpcore/_exceptions.py", line 14, in map_exceptions
airflow-webserver_1       |     raise to_exc(exc)
airflow-webserver_1       | httpcore.ConnectError: [Errno -3] Temporary failure in name resolution
airflow-webserver_1       | 
airflow-webserver_1       | The above exception was the direct cause of the following exception:
airflow-webserver_1       | 
airflow-webserver_1       | Traceback (most recent call last):
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/log/file_task_handler.py", line 508, in _read_from_logs_server
airflow-webserver_1       |     response = _fetch_logs_from_service(url, rel_path)
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/log/file_task_handler.py", line 90, in _fetch_logs_from_service
airflow-webserver_1       |     response = httpx.get(
airflow-webserver_1       |                ^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_api.py", line 189, in get
airflow-webserver_1       |     return request(
airflow-webserver_1       |            ^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_api.py", line 100, in request
airflow-webserver_1       |     return client.request(
airflow-webserver_1       |            ^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_client.py", line 821, in request
airflow-webserver_1       |     return self.send(request, auth=auth, follow_redirects=follow_redirects)
airflow-webserver_1       |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_client.py", line 908, in send
airflow-webserver_1       |     response = self._send_handling_auth(
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_client.py", line 936, in _send_handling_auth
airflow-webserver_1       |     response = self._send_handling_redirects(
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_client.py", line 973, in _send_handling_redirects
airflow-webserver_1       |     response = self._send_single_request(request)
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_client.py", line 1009, in _send_single_request
airflow-webserver_1       |     response = transport.handle_request(request)
airflow-webserver_1       |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_transports/default.py", line 217, in handle_request
airflow-webserver_1       |     with map_httpcore_exceptions():
airflow-webserver_1       |   File "/usr/local/lib/python3.11/contextlib.py", line 155, in __exit__
airflow-webserver_1       |     self.gen.throw(typ, value, traceback)
airflow-webserver_1       |   File "/home/airflow/.local/lib/python3.11/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
airflow-webserver_1       |     raise mapped_exc(message) from exc
airflow-webserver_1       | httpx.ConnectError: [Errno -3] Temporary failure in name resolution
```

**Reply by potiuk** [reply]
Still the same. DNS has a problem. But I have no idea how to help you in your deployment.

**Reply by potiuk** [reply]
Just googled "Temporary failure in name resolution docker". That's the first SO

https://stackoverflow.com/questions/44761246/temporary-failure-in-name-resolution-errno-3-with-docker

But you can continue googling

**Reply by manel00** [reply]
sorry, no clue where to start to fix it, could you give me an example of how it must be an airflow-host in dockercompose and a airflow-worker-slave please? didnt see that through googling

**Reply by potiuk** [reply]
No idea really.

**Reply by manel00** [reply]
one last question please...

how can i fetch the logs from airflow worker (slave server) to host airflow? they are not sharing by any mount/folder the log, how can I get this?

Thank you!

**Reply by potiuk** [reply]
See https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-architecture.html and linked docs - they describe logging architecture.

**Reply by manel00** [reply]
The thing is, in my airflow worker in the slave server, i just have one service: worker, do I need anyone else? I don't understand that much the graphic to be honest

**Reply by potiuk** [reply]
Up to you - you can share it using shared volumes for example.

**Reply by manel00** [reply]
thats the only way to share the logs between host-airflow and slave-worker-airflow?

**Reply by potiuk** [reply]
That's the one that is used by many. But if you want to devise your own, feel free.

**Reply by potiuk** [reply]
As a deployment manager - you are the one responsible for your deployment. Airlfow is generally agnostic how you deploy it. If you want to share files between two machines, it's up to you how you do it.

**Reply by manel00** [reply]
Yeah I know it, but the thing is, I dont get what is the communication between worker-slave-airflow to airflow-host-main regarding logging stuff... that is what I don't understand, thanks for your interest

**Reply by potiuk** [reply, marked as answer]
There is none. UI can download logs from workers via built-in webserver. All the other synchronization has to be done by deployment manager if you want to share logs between machines

**Reply by potiuk** [reply]
BTW that was your original error. UI wanted to download logs from the workers, but when it tried to resolve their name, it failed with temporary resolution error.  The name of your worker is determined by running https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#hostname-callable - this is what worker calls to store the name of the worker in the DB and this is the name, that is used by webserver (not host) to pull the logs to your UI. And it seems that you get temporary resolution error from your DNS when webserver tries to do it.

**Reply by potiuk** [reply]
![image](https://github.com/apache/airflow/assets/595491/eb04c4b8-2bb0-428c-81b3-a3bfbc712d2d)

**Reply by manel00** [reply]
> There is none. UI can download logs from workers via built-in webserver. All the other synchronization has to be done by deployment manager if you want to share logs between machines

Could you please explain better what is "built-in webserver" and "has to be done by deployment manager" with an example please? I am very lost regarding to this subject... I would appreciate that

**Reply by potiuk** [reply]
Unfortunately I do not have all the time of the world to explain the details to eveyrone,. I think you need to read code and tryu understand it if you need. I am helping here in my free time, read the docs, if the doc is not enough -  look at the code if you need to understand things - this is an open-source project used by 10s of thousands of people and my time is better spend on fixing things and maintaining airflow, sorry, but you need to do self-exploration based on the information I already provided.
